### PR TITLE
Implement MAG_DESTROY and MAG_EJECT [RDY]

### PR DESCRIPTION
**Pull request by mugling**
`MAG_DESTROY` is intended for ammo belts and replaces the old behavior which checked for `NO_UNLOAD`. Implemented so that we can have `NO_UNLOAD` cylinders of acetylene for use with the acetylene torch which obviously shouldn't be destroyed when empty.

`MAG_EJECT` provides the previously requested functionality for automatically ejecting an empty magazine/clip. The M1 garand is an ideal demonstration item and making en-bloc clips craftable is a nice balance point.

Combined these two changes bring us closer to implementing @chaosvolt's reusable ammo belts.

Documentation in #15099

### CODE REVIEW

**Comment by chaosvolt** [comment]
Ah, interesting. I was initially worried that `MAG_DESTROY` would entirely preclude the idea of non-disintegrating belts, but that would be only if the flag was implemented on the weapon. The way you have it, implemented on the magazine, appears to be the saner choice. XP

File: data/json/items/magazines.json [hunk 1]
```diff
@@ -1,5 +1,21 @@
 [
     {   "type" : "MAGAZINE",
+        "id" : "garandclip",
+        "price" : 600,
+        "name" : "M1 Garand Clip",
+        "symbol" : "#",
+        "color" : "ltgray",
+        "description" : "A light-weight en-bloc clip for use with the M1 garand that will be ejected with a distinctive ping sound after the last round is fired.",
```
Comment (chaosvolt) [inline data/json/items/magazines.json]: If I recall, amusing discussion has occurred in the past regarding the noises an M1 would make. In a nutstall, while the pinging is due to the en-bloc clip landing on a hard surface, there's also a hollow ringing of sorts as the clip is ejected. o3o

Comment (mugling) [inline data/json/items/magazines.json]: I'm open to other suggestions if you have better wording?

Comment (Rivet-the-Zombie) [inline data/json/items/magazines.json]: I'd say drop the description of the noise entirely and just describe the item's appearance.

Comment (mugling) [inline data/json/items/magazines.json]: Ok. I don't really want to otherwise hold up this PR ammo belts depend upon `MAG_DESTROY` to correctly function

**Comment by kevingranade** [comment]
Not a hold-up for this PR, but are there such things as disintegrating belts that you can just re-link around new ammo?  Seems like it'd be nice to leave a big pile of casings and belt links, then hoover them up for re-creation later.

**Comment by chaosvolt** [comment]
Not sure. As I've mentioned, there are non-disintegrating belts that would be a possible alternative.

File: src/item.cpp [hunk 2]
```diff
@@ -4029,11 +4029,18 @@ bool item::ammo_consume( int qty ) {
...
-                return mag == &e;
-            } ) );
+        auto res = mag->ammo_consume( qty, pos );
+        if( res && ammo_remaining() == 0 ) {
+            if( mag->has_flag( "MAG_DESTROY" ) ) {
+                contents.erase( std::remove_if( contents.begin(), contents.end(), [&mag]( const item& e ) {
+                    return mag == &e;
+                } ) );
+            } else if ( mag->has_flag( "MAG_EJECT" ) ) {
+                g->m.add_item( pos, *mag );
```
Comment (kevingranade) [inline src/item.cpp]: Note for later, might be worthwhile to add a map method like drop_near( pos, item, range ) that finds an open nearby square and drops the item there, good for this and shell casings, and probably some other things.

**Pull request merged**